pkg/cluster/cluster.go

```diff
@@ -251,41 +251,24 @@ func (c *Cluster) run(stopC <-chan struct{}, wg *sync.WaitGroup) {
 				c.logger.Infof("skip reconciliation: running (%v), pending (%v)", k8sutil.GetPodNames(running), k8sutil.GetPodNames(pending))
 				continue
 			}
-			if len(running) == 0 {
-				c.logger.Warningf("all etcd pods are dead. Trying to recover from a previous backup")
-				rerr = c.disasterRecovery(nil)
-				if rerr != nil {
-					c.logger.Errorf("fail to do disaster recovery: %v", rerr)
-				}
-				// On normal recovery case, we need backoff. On error case, this could be either backoff or leading to cluster delete.
-				break
-			}
 
-			// TODO: The case "c.members = nil" only happens on creating seed member.
-			//       We should just set the member directly if successful.
-			if rerr != nil || c.members == nil {
-				rerr = c.updateMembers(podsToMemberSet(running, c.cluster.Spec.SelfHosted))
-				if rerr != nil {
-					c.logger.Errorf("failed to update members: %v", rerr)
-					break
-				}
-			}
-			rerr = c.reconcile(running)
+			// TODO: The case "c.members = nil" only happens on creating seed member. We should just set the member directly if successful.
+			needUpdateMember := rerr != nil || c.members == nil
```

Comment:
But we need to centralize error. Actually we just fix a bug when combining them into a func (we could dismiss the error before).

How about change it to a closure.